### PR TITLE
Fixes to the deprecated package for validation

### PR DESCRIPTION
**Pull request by omarisgreat**
Minimal changes only, to pass validation (as the target is the deprecated IG)

### CODE REVIEW

File: SearchParameter/SearchParameter-immunization-procedure-code.json [hunk 1]
```diff
@@ -13,6 +13,6 @@
       "base" : [
       "Immunization"],
       "type" : "token",
-      "expression" : "Immunization.extension.where(url='https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-VaccinationProcedure').valueCodeableConcept",
+      "expression" : "Immunization.extension.where(url='https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-VaccinationProcedure').value",
```
Comment (omarisgreat) [inline SearchParameter/SearchParameter-immunization-procedure-code.json:16]: Removed CodeableConcept but should resolve to the correct type given the filtered extension only allows one type

File: SearchParameter/SearchParameter-patient-identifier.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
...
+      "expression" : "AllergyIntolerance.patient.identifier | CarePlan.subject.identifier | CareTeam.subject.identifier | ClinicalImpression.subject.identifier | Composition.subject.identifier | Condition.subject.identifier | Consent.patient.identifier | DetectedIssue.patient.identifier | DeviceRequest.subject.identifier | DeviceUseStatement.subject.identifier | DiagnosticReport.subject.identifier | DocumentManifest.subject.identifier | DocumentReference.subject.identifier | Encounter.subject.identifier | EpisodeOfCare.patient.identifier | FamilyMemberHistory.patient.identifier | Flag.subject.identifier | Goal.subject.identifier | ImagingStudy.subject.identifier | Immunization.patient.identifier | List.subject.identifier | MedicationAdministration.subject.identifier | MedicationDispense.subject.identifier | MedicationRequest.subject.identifier | MedicationStatement.subject.identifier | NutritionOrder.patient.identifier | Observation.subject.identifier | Procedure.subject.identifier | RiskAssessment.subject.identifier | ServiceRequest.subject.identifier | SupplyDelivery.patient.identifier | VisionPrescription.patient.identifier",
```
Comment (omarisgreat) [inline SearchParameter/SearchParameter-patient-identifier.json:46]: Removed resolves as cannot resolve on an identifier (only on a reference)

File: StructureDefinition/Extension-NHSDigital-Provenance-agent.StructureDefinition.xml [hunk 3]
```diff
@@ -95,7 +95,6 @@
         <severity value="error" />
         <human value="SDS Role Profile Id must be 12 digits" />
         <expression value="identifier.where(system='https://fhir.nhs.uk/Id/sds-role-profile-id').exists().not() or (identifier.where(system='https://fhir.nhs.uk/Id/sds-role-profile-id').exists()  and identifier.where(system='https://fhir.nhs.uk/Id/sds-role-profile-id').value.matches('^[0-9]{12}$'))" />
-        <source value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-PractitionerRole" />
```
Comment (omarisgreat) [inline StructureDefinition/Extension-NHSDigital-Provenance-agent.StructureDefinition.xml:98]: Removed sources as constraints cannot be imported from other structuredefinitions, only inherited, in which case they do not appear in the SD. removal of the source means this constraint is defined as a new constraint for the profile. This change would not have an impact on validation.

**Review by ryma2fhir — CHANGES_REQUESTED** [review]
Suggest upping version patch to denote bug fixes and change date to suit, but apart from that all looks good

**Comment by omarisgreat** [comment]
There weren't any versions or dates on the structure definition so I've added them, let me know if you would rather these were left out (additionally, the version on the SearchParameters appear to be the FHIR version, but I've just upped the patch number rather than try to change these further)

**Review by ryma2fhir — APPROVED** [review]

**Pull request merged**